### PR TITLE
`--passwordenv` doesn't work for for unlock

### PR DESCRIPTION
**Pull request by Philser**
Add options `--passwordenv` and `--passwordfile` to the `unlock` command

Fixes https://github.com/bitwarden/cli/issues/335

### CODE REVIEW

File: src/commands/unlock.command.ts [hunk 1]
```diff
@@ -14,23 +14,33 @@ import { PasswordVerificationRequest } from 'jslib-common/models/request/passwor
...
-        if (password == null || password === '') {
-            return Response.badRequest('Master password is required.');
+        if ((password == null || password === '')) {
+            if (options.passwordfile) {
+                password = await NodeUtils.readFirstLine(options.passwordfile);
+            } else if (options.passwordenv) {
+                if (process.env[options.passwordenv]) {
+                    password = process.env[options.passwordenv];
+                } else {
+                    return Response.badRequest(`Env variable ${options.passworden} not set.`);
```
[the remark this answers is not in the document]
Comment (MGibson1) [reply src/commands/unlock.command.ts:32]: This is also a small change in behavior from `bw login` I'd like both to work the same. I think I prefer gracefully falling back to user input, if available.

**Review by MGibson1 — CHANGES_REQUESTED** [review]
Thanks for taking this on 👍.

I'd like to reach a consensus on behavior and match the two commands, `bw login` and `bw unlock`. Right now I'm leaning toward graceful fallback, since you can set env variables to handle the `--nointeraction` option environment-wide.

File: src/commands/unlock.command.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Philser) [reply src/commands/unlock.command.ts:32]: Reason why I build that in was that when providing a invalid env var (typo, not set yet) it just prompts you for a password without acknowledging that you provided the passwordenv parameter. Which felt kind of confusing to me. 
But a graceful fallback is probably the better option. 
Would a logging message be proper here, e.g. "could not find env var"?

[the remark this answers is not in the document]
Comment (MGibson1) [reply src/commands/unlock.command.ts:32]: That sounds good. Please be sure to use ConsoleLogService so we can override the output if necessary.

**Comment by Philser** [comment]
@MGibson1 Added a logging message and restored the fallback behaviour. Shall I add the same behaviour to the login command (in the baselib)?

Also, as a note, while --passwordenv has a fallback if the env does not exist, --passwordfile just crashes if an invalid file is provided. Is this intended/desired behaviour?

**Comment by MGibson1** [comment]
Yeah, the `passwordfile` behavior should probably change too, but let's save it for another PR.

Do you mean the warning message for `passwordenv`? Yes please 😄

**Review by MGibson1 — APPROVED** [review]
Looks good, thanks!

**Comment by MGibson1** [comment]
@bitwarden/dept-qa Can I get a go-ahead to merge in for next release?

**Comment by kendratodd** [comment]
@MGibson1 👍.  Thanks for the heads-up. I surfaced this to the rest of the team, and we'll validate once we're past the current release.

**Comment by MGibson1** [comment]
fixes #335

**Pull request merged**